Adding info on setting Kuryr configuration options (#27508)
openshift/openshift-docs: modules/installation-osp-kuryr-port-pools.adoc

// Module included in the following assemblies:
//
// * installing/installing_openstack/installing-openstack-installer-kuryr.adoc
// * installing/installing_openstack/installing-openstack-user-kuryr.adoc
// * post_installation_configuration/network-configuration.adoc

[id="installation-osp-kuryr-port-pools_{context}"]
= Kuryr ports pools

A Kuryr ports pool maintains a number of ports on standby for pod creation.

Keeping ports on standby minimizes pod creation time. Without ports pools,
Kuryr must explicitly request port creation or deletion every time a pod is
created or deleted.

> The Neutron ports that Kuryr uses are created in subnets that are tied to namespaces. These ports are attached as subports to trunk ports: the {product-title} cluster node ports.

Author: All the OpenShift nodes are VMs. It's primary interface is a port. With Kuryr that port has a Neutron "trunk" created.  All the pods ports are a subports of a trunk port of the VM they're running on.

> The Neutron ports that Kuryr uses are created in subnets that are tied to namespaces. These ports are attached as subports to trunk ports: the {product-title} cluster node ports.

Reviewer: How about:
"These ports are attached as subports to the trunk ports that Neutron creates for pods."

> The Neutron ports that Kuryr uses are created in subnets that are tied to namespaces. These ports are attached as subports to trunk ports: the {product-title} cluster node ports.

Author: I rephrased it differently, please check if it makes more sense now.

> The Neutron ports that Kuryr uses are created in subnets that are tied to namespaces. These ports are attached as subports to trunk ports: the {product-title} cluster node ports.

Reviewer: Way better. Thanks!

Because Kuryr keeps each namespace in a separate subnet, a separate ports pool
is maintained for each namespace-worker pair.

> Before triggering the installation you can set the following parameters in the
> `cluster-network-03-config.yml` manifest file to configure ports pool behavior:

Reviewer: ```suggestion
Prior to installing a cluster, you can set the following parameters in the `cluster-network-03-config.yml` manifest file to configure ports pool behavior:
```

> * The `enablePortPoolsPrepopulation` parameter allows you to enable pool prepopulation,
>   which forces Kuryr to add ports to the pool when it is created, such as when a
>   new host is added, or a new namespace is created. The default value is `false`.

Reviewer: ```suggestion
* The `enablePortPoolsPrepopulation` parameter controls pool prepopulation, which forces Kuryr to add ports to the pool when it is created, such as when a new host is added, or a new namespace is created. The default value is `false`.
```

Does "it" here mean "the pool"? 

Option: if there are only a handful of situations where ports are added, rephrase to something like:

"which forces Kuryr to add ports to the pool when a new host is added, a new namespace is created, or [other thing]. The default value is `false`."

Author: Yup, it's pool, this makes sense.

> * The `poolMinPorts` parameter is the minimum number of free ports that are kept in
>   the pool. The default value is `1`.

Reviewer: ```suggestion
* The `poolMinPorts` parameter is the minimum number of free ports that are kept in the pool. The default value is `1`.
```

> * The `poolMaxPorts` parameter is the maximum number of free ports that are kept in the pool.
>   A value of `0` disables that upper bound; this is the default setting. If your OpenStack port quota is
>   low, or you have a limited number of IP addresses on the pod network,
>   consider setting this option to ensure that unneeded ports are deleted.

Reviewer: ```suggestion
* The `poolMaxPorts` parameter is the maximum number of free ports that are kept in the pool. A value of `0` disables that upper bound. This is the default setting.
+
If your OpenStack port quota is low, or you have a limited number of IP addresses on the pod network, consider setting this option to ensure that unneeded ports are deleted.
```
